sessions.md

```diff
@@ -0,0 +1,210 @@
+on the browser.
+
+**NOTE**: Disabling this option is a potential security issue, as your access
+tokens could be hijacked via XSS.
+
+<a href="#top">Back to Top</a>
+
+
+#### <a name="secure"></a> secure
+
+This option determines whether or not our cookies can be set over a non-https
+connection.  By default, this option is null, which means we will attempt to
+inspect the protocol of the incoming request, and make a decision of what to do
+automatically.
+
+Defining this value forces the `secure` mode to be either enabled or disabled
+all the time.
+
+**NOTE**: Disabling this option is a potential security issue, as setting
+cookies over a non-https connection is ripe for traffic inspection attacks.
+
+<a href="#top">Back to Top</a>
+
+
+#### <a name="path"></a> path
+
+This option determines what URI paths will be able to read our cookies.  By
+default, this is set to the root path (`/`), meaning our cookies will be
+available to every URI on the web server.
+
+<a href="#top">Back to Top</a>
+
+
+#### <a name="domain"></a> domain
+
+This option determines what domain(s), can read this cookie value.  By default,
+this is `null`, meaning only the current domain where the cookie is set can use
+read this cookie value.
+
+The only time you might want to set this value is if you want to allow
+subdomains to access your cookies as well.
+
+<a href="#top">Back to Top</a>
+
+
+#### <a name="validation-strategy"></a> validationStrategy
+
+Determines how we validate an access token.  There are three strategies:
+
+* **local** - only verify the signature, expiration, and issuer
+* **stormpath** - always validate against Stormpath, for additional checks
+such as Account and Application status.  This is the default, but the most
+consistent (will always be aware of Stormpath resource statuses).
+* **random** - 50/50 chance, per request, of **local** or **stormpath** being used.
+This balances speed with consistency.
+
+<a href="#top">Back to Top</a>
+
+
+## <a name="incoming-request-handling"></a> Incoming Request Handling
+
+For any request that comes into the framework, we attempt to validate the access
+token, and, if invalid, we'll attempt to extend the session by getting and
+setting a new access token (using the refresh token).
+
+If the access token is valid, we'll simply fetch the account object and attach
+it to the request context.
+
+If for any reason, the refresh token is invalid, we'll immediately set the
+`accessToken` and `refreshToken` cookie values to empty strings, ensuring they
+are removed from the browser.
+
+If the developer has declared, "loginRequired" for this route, and if the
+requested content type is `json`, we'll return a 401 response -- otherwise if
+the content type  is HTML, we should 302 Redirect to the login view.  This flow
+ensures that both client side JS frameworks and server side frameworks work as
+expected.
+
+
+<a href="#top">Back to Top</a>
+
+
+## <a name="me-handling"></a> /me Handling
```

Comment:
Why this is required by all integrations?  App devs already have access to the `Account` in the request, and can return any type of response (HTML, JSON) as they see fit for whatever their UI requires.  Thoughts?